File: 1668/go/1668.go
func maxRepeating(sequence string, word string) int {
    sl := len(sequence)
    wl := len(word)

    // make dp
    dp := make([]int, sl+1)

    max := 0

    for i := 1; i < sl+1; i++ {
        if i >= wl {
            if sequence[i-wl:i] == word {
                dp[i] = dp[i-wl] + 1
                if max < dp[i]{
                    max = dp[i]
                }
            }
        }
    }

    return max
}

File: 1668/py/1668.py
class Solution:
    def maxRepeating(self, sequence: str, word: str) -> int:
        sl = len(sequence)
        wl = len(word)
        # make dp
        dp = [0 for _ in range(sl+1)]

        # record
        max = 0

        for i in range(1, sl+1):
            if i >= wl :
                if sequence[(i-wl):i] == word :
                    dp[i] = dp[i-wl] + 1
                    if max < dp[i]:
                        max = dp[i]
        return max